src/d3a/blockchain/contracts/ink/lib.rs

```diff
@@ -27,6 +27,7 @@ mod trade_storage {
     pub struct TradeStorage {
         /// The trades stored in the contract.
         trades: StorageHashMap<Hash, (AccountId, AccountId, Balance, i8)>,
+        area_market_id: StorageHashMap<u128, (u128, u32)>,
```

Comment:
Trivial, but I think this is better named as simulation_market_id or simulation_area_id, and stick to either using the term `market` or `area` for the whole smart contract (I do not mind which one of the 2, the term has to be consistent though).

Reply:
Possibly better to use DoubleStorage instead of Storage, as Storage is one-to-one kay-value mapping. https://substrate.dev/docs/en/knowledgebase/runtime/storage

Reply:
As discussed offline `DoubleStorage ` is only for runtime storage and not for ink, I will close this  PR now and will look into the possibility of either importing runtime trade storage in ink or to implement via pallet contract.